[BUG] networkx tests failed with crash

**Describe the bug**

```python
2023-06-06T05:18:40.5733303Z 
2023-06-06T05:18:40.5733743Z =============================== warnings summary ===============================
2023-06-06T05:18:40.5734545Z ../../../.local/lib/python3.8/site-packages/graphscope/nx/conftest.py:47
2023-06-06T05:18:40.5735523Z   /home/runner/.local/lib/python3.8/site-packages/graphscope/nx/conftest.py:47: PytestUnknownMarkWarning: Unknown pytest.mark.timeout - is this a typo?  You can register custom marks to avoid this warning - for details, see https://docs.pytest.org/en/stable/how-to/mark.html
2023-06-06T05:18:40.5736097Z     item.add_marker(pytest.mark.timeout(600))
2023-06-06T05:18:40.5736271Z 
2023-06-06T05:18:40.5736741Z .local/lib/python3.8/site-packages/graphscope/nx/tests/test_transformation.py::TestGraphTransformation::test_empty_nx_to_gs
2023-06-06T05:18:40.5737415Z .local/lib/python3.8/site-packages/graphscope/nx/tests/test_transformation.py::TestDigraphTransformation::test_empty_nx_to_gs
2023-06-06T05:18:40.5738114Z   /home/runner/.local/lib/python3.8/site-packages/graphscope/framework/utils.py:536: DeprecationWarning: `np.bool8` is a deprecated alias for `np.bool_`.  (Deprecated NumPy 1.24)
2023-06-06T05:18:40.5738522Z     np.bool8: graph_def_pb2.BOOL,
2023-06-06T05:18:40.5738664Z 
2023-06-06T05:18:40.5739064Z .local/lib/python3.8/site-packages/graphscope/nx/tests/test_transformation.py::TestGraphTransformation::test_multi_label_gs_to_nx
2023-06-06T05:18:40.5739750Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-10071
2023-06-06T05:18:40.5740127Z   
2023-06-06T05:18:40.5740315Z   Traceback (most recent call last):
2023-06-06T05:18:40.5740611Z     File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
2023-06-06T05:18:40.5740996Z       self.run()
2023-06-06T05:18:40.5741236Z     File "/usr/lib/python3.8/threading.py", line 870, in run
2023-06-06T05:18:40.5741516Z       self._target(*self._args, **self._kwargs)
2023-06-06T05:18:40.5741975Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/graph.py", line 386, in _del
2023-06-06T05:18:40.5742288Z       graph.cache.shutdown()
2023-06-06T05:18:40.5742737Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 257, in shutdown
2023-06-06T05:18:40.5743045Z       future.result()
2023-06-06T05:18:40.5743324Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
2023-06-06T05:18:40.5743589Z       return self.__get_result()
2023-06-06T05:18:40.5743881Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
2023-06-06T05:18:40.5744163Z       raise self._exception
2023-06-06T05:18:40.5744497Z     File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
2023-06-06T05:18:40.5744811Z       result = self.fn(*self.args, **self.kwargs)
2023-06-06T05:18:40.5745292Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 361, in _get_node_attr_cache
2023-06-06T05:18:40.5745655Z       node_attr_cache = msgpack.load(fp, use_list=False)
2023-06-06T05:18:40.5746096Z     File "/home/runner/.local/lib/python3.8/site-packages/msgpack/__init__.py", line 49, in unpack
2023-06-06T05:18:40.5746398Z       return unpackb(data, **kwargs)
2023-06-06T05:18:40.5746689Z     File "msgpack/_unpacker.pyx", line 201, in msgpack._cmsgpack.unpackb
2023-06-06T05:18:40.5747028Z   msgpack.exceptions.ExtraData: unpack(b) received extra data.
2023-06-06T05:18:40.5747271Z   
2023-06-06T05:18:40.5747557Z     warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
2023-06-06T05:18:40.5747848Z 
2023-06-06T05:18:40.5748249Z .local/lib/python3.8/site-packages/graphscope/nx/tests/test_transformation.py::TestGraphTransformation::test_str_oid_gs_to_nx
2023-06-06T05:18:40.5748934Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-10115
2023-06-06T05:18:40.5749308Z   
2023-06-06T05:18:40.5749498Z   Traceback (most recent call last):
2023-06-06T05:18:40.5749780Z     File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
2023-06-06T05:18:40.5750080Z       self.run()
2023-06-06T05:18:40.5750329Z     File "/usr/lib/python3.8/threading.py", line 870, in run
2023-06-06T05:18:40.5750594Z       self._target(*self._args, **self._kwargs)
2023-06-06T05:18:40.5751035Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/graph.py", line 386, in _del
2023-06-06T05:18:40.5751353Z       graph.cache.shutdown()
2023-06-06T05:18:40.5751787Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 257, in shutdown
2023-06-06T05:18:40.5752090Z       future.result()
2023-06-06T05:18:40.5752370Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
2023-06-06T05:18:40.5752632Z       return self.__get_result()
2023-06-06T05:18:40.5752923Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
2023-06-06T05:18:40.5753193Z       raise self._exception
2023-06-06T05:18:40.5753475Z     File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
2023-06-06T05:18:40.5753767Z       result = self.fn(*self.args, **self.kwargs)
2023-06-06T05:18:40.5754576Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 361, in _get_node_attr_cache
2023-06-06T05:18:40.5754949Z       node_attr_cache = msgpack.load(fp, use_list=False)
2023-06-06T05:18:40.5755373Z     File "/home/runner/.local/lib/python3.8/site-packages/msgpack/__init__.py", line 49, in unpack
2023-06-06T05:18:40.5755686Z       return unpackb(data, **kwargs)
2023-06-06T05:18:40.5755987Z     File "msgpack/_unpacker.pyx", line 201, in msgpack._cmsgpack.unpackb
2023-06-06T05:18:40.5756399Z   msgpack.exceptions.ExtraData: unpack(b) received extra data.
2023-06-06T05:18:40.5756638Z   
2023-06-06T05:18:40.5756931Z     warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
2023-06-06T05:18:40.5757162Z 
2023-06-06T05:18:40.5757564Z .local/lib/python3.8/site-packages/graphscope/nx/tests/test_transformation.py::TestGraphTransformation::test_multiple_sessions
2023-06-06T05:18:40.5758249Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-10136
2023-06-06T05:18:40.5758611Z   
2023-06-06T05:18:40.5758799Z   Traceback (most recent call last):
2023-06-06T05:18:40.5759083Z     File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
2023-06-06T05:18:40.5759335Z       self.run()
2023-06-06T05:18:40.5759583Z     File "/usr/lib/python3.8/threading.py", line 870, in run
2023-06-06T05:18:40.5759909Z       self._target(*self._args, **self._kwargs)
2023-06-06T05:18:40.5760354Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/graph.py", line 386, in _del
2023-06-06T05:18:40.5760674Z       graph.cache.shutdown()
2023-06-06T05:18:40.5761113Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 257, in shutdown
2023-06-06T05:18:40.5761407Z       future.result()
2023-06-06T05:18:40.5761683Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
2023-06-06T05:18:40.5761951Z       return self.__get_result()
2023-06-06T05:18:40.5762233Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
2023-06-06T05:18:40.5762504Z       raise self._exception
2023-06-06T05:18:40.5762784Z     File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
2023-06-06T05:18:40.5763144Z       result = self.fn(*self.args, **self.kwargs)
2023-06-06T05:18:40.5763613Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 361, in _get_node_attr_cache
2023-06-06T05:18:40.5763989Z       node_attr_cache = msgpack.load(fp, use_list=False)
2023-06-06T05:18:40.5764423Z     File "/home/runner/.local/lib/python3.8/site-packages/msgpack/__init__.py", line 49, in unpack
2023-06-06T05:18:40.5764719Z       return unpackb(data, **kwargs)
2023-06-06T05:18:40.5765013Z     File "msgpack/_unpacker.pyx", line 201, in msgpack._cmsgpack.unpackb
2023-06-06T05:18:40.5765412Z   msgpack.exceptions.ExtraData: unpack(b) received extra data.
2023-06-06T05:18:40.5765654Z   
2023-06-06T05:18:40.5765943Z     warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
2023-06-06T05:18:40.5766178Z 
2023-06-06T05:18:40.5766592Z .local/lib/python3.8/site-packages/graphscope/nx/tests/test_transformation.py::TestDigraphTransformation::test_multi_label_gs_to_nx
2023-06-06T05:18:40.5767287Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-10537
2023-06-06T05:18:40.5767647Z   
2023-06-06T05:18:40.5767835Z   Traceback (most recent call last):
2023-06-06T05:18:40.5768122Z     File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
2023-06-06T05:18:40.5768365Z       self.run()
2023-06-06T05:18:40.5768607Z     File "/usr/lib/python3.8/threading.py", line 870, in run
2023-06-06T05:18:40.5768880Z       self._target(*self._args, **self._kwargs)
2023-06-06T05:18:40.5769324Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/graph.py", line 386, in _del
2023-06-06T05:18:40.5769638Z       graph.cache.shutdown()
2023-06-06T05:18:40.5770079Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 257, in shutdown
2023-06-06T05:18:40.5770385Z       future.result()
2023-06-06T05:18:40.5770649Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
2023-06-06T05:18:40.5770919Z       return self.__get_result()
2023-06-06T05:18:40.5775223Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
2023-06-06T05:18:40.5775676Z       raise self._exception
2023-06-06T05:18:40.5775952Z     File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
2023-06-06T05:18:40.5776251Z       result = self.fn(*self.args, **self.kwargs)
2023-06-06T05:18:40.5776800Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 361, in _get_node_attr_cache
2023-06-06T05:18:40.5777162Z       node_attr_cache = msgpack.load(fp, use_list=False)
2023-06-06T05:18:40.5777602Z     File "/home/runner/.local/lib/python3.8/site-packages/msgpack/__init__.py", line 49, in unpack
2023-06-06T05:18:40.5777905Z       return unpackb(data, **kwargs)
2023-06-06T05:18:40.5778194Z     File "msgpack/_unpacker.pyx", line 201, in msgpack._cmsgpack.unpackb
2023-06-06T05:18:40.5778549Z   msgpack.exceptions.ExtraData: unpack(b) received extra data.
2023-06-06T05:18:40.5778800Z   
2023-06-06T05:18:40.5779131Z     warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
2023-06-06T05:18:40.5779369Z 
2023-06-06T05:18:40.5779808Z .local/lib/python3.8/site-packages/graphscope/nx/tests/test_transformation.py::TestDigraphTransformation::test_str_oid_gs_to_nx
2023-06-06T05:18:40.5780582Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-10581
2023-06-06T05:18:40.5780951Z   
2023-06-06T05:18:40.5781142Z   Traceback (most recent call last):
2023-06-06T05:18:40.5781437Z     File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
2023-06-06T05:18:40.5781686Z       self.run()
2023-06-06T05:18:40.5781928Z     File "/usr/lib/python3.8/threading.py", line 870, in run
2023-06-06T05:18:40.5782200Z       self._target(*self._args, **self._kwargs)
2023-06-06T05:18:40.5782692Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/graph.py", line 386, in _del
2023-06-06T05:18:40.5783011Z       graph.cache.shutdown()
2023-06-06T05:18:40.5783452Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 257, in shutdown
2023-06-06T05:18:40.5783749Z       future.result()
2023-06-06T05:18:40.5784028Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
2023-06-06T05:18:40.5784308Z       return self.__get_result()
2023-06-06T05:18:40.5784654Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
2023-06-06T05:18:40.5784920Z       raise self._exception
2023-06-06T05:18:40.5785198Z     File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
2023-06-06T05:18:40.5785497Z       result = self.fn(*self.args, **self.kwargs)
2023-06-06T05:18:40.5785963Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 361, in _get_node_attr_cache
2023-06-06T05:18:40.5786335Z       node_attr_cache = msgpack.load(fp, use_list=False)
2023-06-06T05:18:40.5786775Z     File "/home/runner/.local/lib/python3.8/site-packages/msgpack/__init__.py", line 49, in unpack
2023-06-06T05:18:40.5787078Z       return unpackb(data, **kwargs)
2023-06-06T05:18:40.5787362Z     File "msgpack/_unpacker.pyx", line 201, in msgpack._cmsgpack.unpackb
2023-06-06T05:18:40.5787711Z   msgpack.exceptions.ExtraData: unpack(b) received extra data.
2023-06-06T05:18:40.5787965Z   
2023-06-06T05:18:40.5788244Z     warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
2023-06-06T05:18:40.5788469Z 
2023-06-06T05:18:40.5788884Z .local/lib/python3.8/site-packages/graphscope/nx/tests/test_transformation.py::TestDigraphTransformation::test_error_on_wrong_nx_type
2023-06-06T05:18:40.5789571Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/threadexception.py:73: PytestUnhandledThreadExceptionWarning: Exception in thread Thread-10602
2023-06-06T05:18:40.5789933Z   
2023-06-06T05:18:40.5790111Z   Traceback (most recent call last):
2023-06-06T05:18:40.5790404Z     File "/usr/lib/python3.8/threading.py", line 932, in _bootstrap_inner
2023-06-06T05:18:40.5790719Z       self.run()
2023-06-06T05:18:40.5790956Z     File "/usr/lib/python3.8/threading.py", line 870, in run
2023-06-06T05:18:40.5791224Z       self._target(*self._args, **self._kwargs)
2023-06-06T05:18:40.5791666Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/graph.py", line 386, in _del
2023-06-06T05:18:40.5791973Z       graph.cache.shutdown()
2023-06-06T05:18:40.5792405Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 257, in shutdown
2023-06-06T05:18:40.5792715Z       future.result()
2023-06-06T05:18:40.5792981Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 437, in result
2023-06-06T05:18:40.5793254Z       return self.__get_result()
2023-06-06T05:18:40.5793545Z     File "/usr/lib/python3.8/concurrent/futures/_base.py", line 389, in __get_result
2023-06-06T05:18:40.5793821Z       raise self._exception
2023-06-06T05:18:40.5794344Z     File "/usr/lib/python3.8/concurrent/futures/thread.py", line 57, in run
2023-06-06T05:18:40.5794655Z       result = self.fn(*self.args, **self.kwargs)
2023-06-06T05:18:40.5795141Z     File "/home/runner/.local/lib/python3.8/site-packages/graphscope/nx/classes/cache.py", line 361, in _get_node_attr_cache
2023-06-06T05:18:40.5795502Z       node_attr_cache = msgpack.load(fp, use_list=False)
2023-06-06T05:18:40.5795940Z     File "/home/runner/.local/lib/python3.8/site-packages/msgpack/__init__.py", line 49, in unpack
2023-06-06T05:18:40.5796249Z       return unpackb(data, **kwargs)
2023-06-06T05:18:40.5796537Z     File "msgpack/_unpacker.pyx", line 201, in msgpack._cmsgpack.unpackb
2023-06-06T05:18:40.5796886Z   msgpack.exceptions.ExtraData: unpack(b) received extra data.
2023-06-06T05:18:40.5797144Z   
2023-06-06T05:18:40.5797433Z     warnings.warn(pytest.PytestUnhandledThreadExceptionWarning(msg))
2023-06-06T05:18:40.5797720Z 
2023-06-06T05:18:40.5798096Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_function.py::TestFunction::test_info_digraph
2023-06-06T05:18:40.5798771Z   /home/runner/.local/lib/python3.8/site-packages/networkx/classes/tests/test_function.py:281: DeprecationWarning: info is deprecated and will be removed in version 3.0.
2023-06-06T05:18:40.5799129Z   
2023-06-06T05:18:40.5799288Z     info = nx.info(G)
2023-06-06T05:18:40.5799411Z 
2023-06-06T05:18:40.5799822Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_function.py::TestFunction::test_info_digraph
2023-06-06T05:18:40.5800469Z   /home/runner/.local/lib/python3.8/site-packages/networkx/classes/tests/test_function.py:285: DeprecationWarning: info is deprecated and will be removed in version 3.0.
2023-06-06T05:18:40.5800811Z   
2023-06-06T05:18:40.5800974Z     info = nx.info(G, n=1)
2023-06-06T05:18:40.5801104Z 
2023-06-06T05:18:40.5801459Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_function.py::TestFunction::test_info_digraph
2023-06-06T05:18:40.5802078Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/python_api.py:951: DeprecationWarning: info is deprecated and will be removed in version 3.0.
2023-06-06T05:18:40.5802425Z   
2023-06-06T05:18:40.5802593Z     func(*args[1:], **kwargs)
2023-06-06T05:18:40.5802722Z 
2023-06-06T05:18:40.5803085Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestReverseView::test_contains
2023-06-06T05:18:40.5803753Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5804508Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestReverseView::test_contains is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5804933Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5805434Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5805789Z     fixture_result = next(generator)
2023-06-06T05:18:40.5805996Z 
2023-06-06T05:18:40.5806352Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestReverseView::test_iter
2023-06-06T05:18:40.5807012Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5807762Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestReverseView::test_iter is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5808185Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5808654Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5808998Z     fixture_result = next(generator)
2023-06-06T05:18:40.5809149Z 
2023-06-06T05:18:40.5809566Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestReverseView::test_exceptions
2023-06-06T05:18:40.5810237Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5810986Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestReverseView::test_exceptions is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5811410Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5811880Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5812221Z     fixture_result = next(generator)
2023-06-06T05:18:40.5812360Z 
2023-06-06T05:18:40.5812714Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestReverseView::test_subclass
2023-06-06T05:18:40.5813416Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5814163Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestReverseView::test_subclass is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5814588Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5815096Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5815440Z     fixture_result = next(generator)
2023-06-06T05:18:40.5815585Z 
2023-06-06T05:18:40.5815951Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToDirected::test_directed
2023-06-06T05:18:40.5816617Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5817355Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToDirected::test_directed is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5817780Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5818251Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5818592Z     fixture_result = next(generator)
2023-06-06T05:18:40.5818733Z 
2023-06-06T05:18:40.5819091Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToDirected::test_contains
2023-06-06T05:18:40.5819749Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5820483Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToDirected::test_contains is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5820913Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5821430Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5821773Z     fixture_result = next(generator)
2023-06-06T05:18:40.5821917Z 
2023-06-06T05:18:40.5822285Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToDirected::test_pickle
2023-06-06T05:18:40.5822953Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5823680Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToDirected::test_pickle is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5824108Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5824578Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5824962Z     fixture_result = next(generator)
2023-06-06T05:18:40.5825103Z 
2023-06-06T05:18:40.5825476Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToDirected::test_already_directed
2023-06-06T05:18:40.5826145Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5826899Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToDirected::test_already_directed is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5827335Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5827798Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5828137Z     fixture_result = next(generator)
2023-06-06T05:18:40.5828343Z 
2023-06-06T05:18:40.5828703Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToDirected::test_iter
2023-06-06T05:18:40.5829361Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5830128Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToDirected::test_iter is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5830584Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5831058Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5831402Z     fixture_result = next(generator)
2023-06-06T05:18:40.5831537Z 
2023-06-06T05:18:40.5831908Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToUndirected::test_directed
2023-06-06T05:18:40.5832574Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5833327Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToUndirected::test_directed is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5833761Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5834322Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5834668Z     fixture_result = next(generator)
2023-06-06T05:18:40.5834810Z 
2023-06-06T05:18:40.5835179Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToUndirected::test_contains
2023-06-06T05:18:40.5835850Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5836596Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToUndirected::test_contains is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5837093Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5837561Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5837901Z     fixture_result = next(generator)
2023-06-06T05:18:40.5838035Z 
2023-06-06T05:18:40.5838416Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToUndirected::test_already_directed
2023-06-06T05:18:40.5839093Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5839854Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToUndirected::test_already_directed is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5840344Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5840815Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5841152Z     fixture_result = next(generator)
2023-06-06T05:18:40.5841297Z 
2023-06-06T05:18:40.5841662Z .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToUndirected::test_iter
2023-06-06T05:18:40.5842321Z   /home/runner/.local/lib/python3.8/site-packages/_pytest/fixtures.py:902: PytestRemovedIn8Warning: Support for nose tests is deprecated and will be removed in a future release.
2023-06-06T05:18:40.5843048Z   .local/lib/python3.8/site-packages/graphscope/nx/tests/classes/test_graphviews.py::TestToUndirected::test_iter is using nose-specific method: `setup(self)`
2023-06-06T05:18:40.5843464Z   To remove this warning, rename it to `setup_method(self)`
2023-06-06T05:18:40.5843987Z   See docs: https://docs.pytest.org/en/stable/deprecations.html#support-for-tests-written-for-nose
2023-06-06T05:18:40.5844321Z     fixture_result = next(generator)
2023-06-06T05:18:40.5844455Z 
2023-06-06T05:18:40.5844710Z -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
2023-06-06T05:18:40.5845059Z ========== 555 passed, 23 skipped, 25 warnings in 2529.34s (0:42:09) ===========
2023-06-06T05:19:22.3530778Z /home/runner/work/_temp/28480107-f529-4759-b90c-101bdc95ed3a.sh: line 4:  1896 Segmentation fault      (core dumped) python3 -m pytest --exitfirst -s -v $(dirname $(python3 -c "import graphscope; print(graphscope.__file__)"))/nx/tests --ignore=$(dirname $(python3 -c "import graphscope; print(graphscope.__file__)"))/nx/tests/convert
2023-06-06T05:19:22.3596473Z ##[error]Process completed with exit code 139.
2023-06-06T05:19:22.3804753Z Post job cleanup.
2023-06-06T05:19:22.5398168Z [command]/usr/bin/git version
2023-06-06T05:19:22.5483431Z git version 2.40.1
2023-06-06T05:19:22.5537436Z Temporarily overriding HOME='/home/runner/work/_temp/77c92fa5-2fd8-46b1-aad4-87b724ea3e03' before making global git config changes
2023-06-06T05:19:22.5540867Z Adding repository directory to the temporary git global config as a safe directory
2023-06-06T05:19:22.5548097Z [command]/usr/bin/git config --global --add safe.directory /home/runner/work/GraphScope/GraphScope
2023-06-06T05:19:22.5623957Z [command]/usr/bin/git config --local --name-only --get-regexp core\.sshCommand
2023-06-06T05:19:22.5676530Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'core\.sshCommand' && git config --local --unset-all 'core.sshCommand' || :"
2023-06-06T05:19:22.6006192Z [command]/usr/bin/git config --local --name-only --get-regexp http\.https\:\/\/github\.com\/\.extraheader
2023-06-06T05:19:22.6033120Z http.https://github.com/.extraheader
2023-06-06T05:19:22.6045161Z [command]/usr/bin/git config --local --unset-all http.https://github.com/.extraheader
2023-06-06T05:19:22.6079962Z [command]/usr/bin/git submodule foreach --recursive sh -c "git config --local --name-only --get-regexp 'http\.https\:\/\/github\.com\/\.extraheader' && git config --local --unset-all 'http.https://github.com/.extraheader' || :"
2023-06-06T05:19:22.6654007Z Cleaning up orphan processes
```

**Additional context**

See also: https://github.com/alibaba/GraphScope/actions/runs/5184446534/jobs/9343725706

--- a/analytical_engine/core/fragment/dynamic_fragment.h
+++ b/analytical_engine/core/fragment/dynamic_fragment.h
@@ -1529,7 +1529,7 @@ class DynamicFragmentMutator {
       v_fid = partitioner.GetPartitionId(oid);
       if (modify_type == rpc::NX_ADD_NODES) {
         vm_ptr_->AddVertex(std::move(oid), gid);
-        if (!v_data.Empty()) {
+        if (v_data.IsObject() && !v_data.GetObject().ObjectEmpty()) {
           for (const auto& prop : v_data.GetObject()) {
             if (!fragment_->schema_["vertex"].HasMember(prop.name)) {
               dynamic::Value key(prop.name);
@@ -1606,7 +1606,7 @@ class DynamicFragmentMutator {
                                                   std::move(empty_data));
           }
         }
-        if (!e_data.Empty()) {
+        if (e_data.IsObject() && !e_data.GetObject().ObjectEmpty()) {
           for (const auto& prop : e_data.GetObject()) {
             if (!fragment_->schema_["edge"].HasMember(prop.name)) {
               dynamic::Value key(prop.name);

--- a/analytical_engine/core/fragment/fragment_reporter.h
+++ b/analytical_engine/core/fragment/fragment_reporter.h
@@ -105,7 +105,6 @@ class DynamicFragmentReporter : public grape::Communicator {
       }
       break;
     }
-
     case rpc::HAS_EDGE: {
       BOOST_LEAF_AUTO(edge_in_json, params.Get<std::string>(rpc::EDGE));
       dynamic::Value edge;
@@ -761,7 +760,9 @@ class ArrowFragmentReporter<
         }
       }
       // archive the start gid and nodes attribute array.
-      arc << gid << nodes_attr;
+      msgpack::sbuffer sbuf;
+      msgpack::pack(&sbuf, nodes_attr);
+      arc << gid << sbuf;
     }
   }
 
@@ -859,7 +860,9 @@ class ArrowFragmentReporter<
         }
       }
       // archive the start gid and edges attributes array.
-      arc << gid << adj_list;
+      msgpack::sbuffer sbuf;
+      msgpack::pack(&sbuf, adj_list);
+      arc << gid << sbuf;
     }
   }
 

--- a/python/graphscope/client/session.py
+++ b/python/graphscope/client/session.py
@@ -787,6 +787,10 @@ class Session(object):
     def closed(self):
         return self._closed
 
+    @property
+    def disconnected(self):
+        return self._grpc_client is None or self._disconnected
+
     def eager(self):
         return self._config_params["mode"] == "eager"
 

--- a/python/graphscope/framework/app.py
+++ b/python/graphscope/framework/app.py
@@ -352,6 +352,10 @@ class AppDAGNode(DAGNode):
         # add op to dag
         self._session.dag.add_op(self._op)
 
+        # statically create the unload op to prevent a possible segmentation fault
+        # inside the protobuf library.
+        self._unload_op = unload_app(self)
+
     def __repr__(self):
         s = f"graphscope.App <type: {self._app_assets.type}, algorithm: {self._app_assets.algo} "
         s += f"bounded_graph: {str(self._graph)}>"
@@ -420,8 +424,7 @@ class AppDAGNode(DAGNode):
         Returns:
             :class:`graphscope.framework.app.UnloadedApp`: Evaluated in eager mode.
         """
-        op = unload_app(self)
-        return UnloadedApp(self._session, op)
+        return UnloadedApp(self._session, self._unload_op)
 
 
 class App(object):

--- a/python/graphscope/framework/context.py
+++ b/python/graphscope/framework/context.py
@@ -121,6 +121,9 @@ class BaseContextDAGNode(DAGNode):
         self._op = run_app(self._bound_app, *args, **kwargs)
         self._session.dag.add_op(self._op)
 
+        # statically create the unload op
+        self._unload_op = dag_utils.unload_context(self)
+
     def _check_selector(self, selector):
         raise NotImplementedError()
 
@@ -279,8 +282,7 @@ class BaseContextDAGNode(DAGNode):
             pass
 
     def _unload(self):
-        op = dag_utils.unload_context(self)
-        return UnloadedContext(self._session, op)
+        return UnloadedContext(self._session, self._unload_op)
 
 
 class TensorContextDAGNode(BaseContextDAGNode):

--- a/python/graphscope/framework/graph.py
+++ b/python/graphscope/framework/graph.py
@@ -292,6 +292,9 @@ class GraphDAGNode(DAGNode, GraphInterface):
         self._resolve_op(incoming_data)
         self._session.dag.add_op(self._op)
 
+        # statically create the unload op
+        self._unload_op = dag_utils.unload_graph(self)
+
     @property
     def v_labels(self):
         return self._v_labels
@@ -714,8 +717,7 @@ class GraphDAGNode(DAGNode, GraphInterface):
         Returns:
             :class:`graphscope.framework.graph.UnloadedGraph`: Evaluated in eager mode.
         """
-        op = dag_utils.unload_graph(self)
-        return UnloadedGraph(self._session, op)
+        return UnloadedGraph(self._session, self._unload_op)
 
     def project(
         self,

--- a/python/graphscope/nx/classes/cache.py
+++ b/python/graphscope/nx/classes/cache.py
@@ -113,9 +113,8 @@ class Cache:
     def align_node_attr_cache(self):
         """Check and align the node attr cache with node id cache"""
         if self.enable_iter_cache and self.node_attr_align is False:
-            f = self.futures["node_attr"]
-            if f is not None:
-                start_gid, self.node_attr_cache = f.result()
+            if self.futures["node_attr"] is not None:
+                start_gid, self.node_attr_cache = self.futures["node_attr"].result()
                 if start_gid == self.iter_pre_gid:
                     # align to current node_id_cache
                     if self.iter_gid != self.iter_pre_gid:
@@ -129,8 +128,7 @@ class Cache:
     def align_succ_cache(self):
         """Check and align the succ neighbor cache with node id cache"""
         if self.enable_iter_cache and self.succ_align is False:
-            f = self.futures["succ"]
-            start_gid, self.succ_cache = f.result()
+            start_gid, self.succ_cache = self.futures["succ"].result()
             if start_gid == self.iter_pre_gid:
                 if self.iter_gid != self.iter_pre_gid:
                     self._async_fetch_succ_cache(self.iter_gid)
@@ -143,9 +141,8 @@ class Cache:
     def align_succ_attr_cache(self):
         """Check and align the succ neighbor attr cache with node id cache"""
         if self.enable_iter_cache and self.succ_attr_align is False:
-            f = self.futures["succ_attr"]
-            if f is not None:
-                start_gid, self.succ_attr_cache = f.result()
+            if self.futures["succ_attr"] is not None:
+                start_gid, self.succ_attr_cache = self.futures["succ_attr"].result()
                 if start_gid == self.iter_pre_gid:
                     if self.iter_gid != self.iter_pre_gid:
                         self._async_fetch_succ_attr_cache(self.iter_gid)
@@ -160,8 +157,7 @@ class Cache:
         if self.enable_iter_cache and self.pred_align is False:
             if self.futures["pred"] is None:
                 self._async_fetch_pred_cache(self.iter_pre_gid)
-            f = self.futures["pred"]
-            start_gid, self.pred_cache = f.result()
+            start_gid, self.pred_cache = self.futures["pred"].result()
             if start_gid == self.iter_pre_gid:
                 if self.iter_gid != self.iter_pre_gid:
                     self._async_fetch_pred_cache(self.iter_gid)
@@ -177,8 +173,7 @@ class Cache:
         if self.enable_iter_cache and self.pred_attr_align is False:
             if self.futures["pred_attr"] is None:
                 self._async_fetch_pred_attr_cache(self.iter_pre_gid)
-            f = self.futures["pred_attr"]
-            start_gid, self.pred_attr_cache = f.result()
+            start_gid, self.pred_attr_cache = self.futures["pred_attr"].result()
             if start_gid == self.iter_pre_gid:
                 if self.iter_gid != self.iter_pre_gid:
                     self._async_fetch_pred_attr_cache(self.iter_gid)
@@ -258,6 +253,9 @@ class Cache:
                 except concurrent.futures.CancelledError:
                     pass
                 future = None
+
+    def shutdown_executor(self):
+        self.executor.shutdown(wait=True)
 
     def clear(self):
         """Clear batch cache and lru cache, reset the status and warmup again"""

--- a/python/graphscope/nx/classes/digraph.py
+++ b/python/graphscope/nx/classes/digraph.py
@@ -29,7 +29,7 @@ from networkx.classes.reportviews import OutDegreeView
 
 from graphscope.client.session import get_default_session
 from graphscope.client.session import get_session_by_id
-from graphscope.framework.dag_utils import copy_graph
+from graphscope.framework import dag_utils
 from graphscope.framework.errors import check_argument
 from graphscope.framework.graph_schema import GraphSchema
 from graphscope.nx import NetworkXError
@@ -295,6 +295,12 @@ class DiGraph(Graph):
         self._saved_signature = self.signature
         self._is_client_view = False
 
+        # statically create the unload op
+        if self.op is None:
+            self._unload_op = None
+        else:
+            self._unload_op = dag_utils.unload_graph(self)
+
     @property
     @clear_mutation_cache
     @patch_docstring(RefDiGraph.adj)
@@ -494,7 +500,7 @@ class DiGraph(Graph):
             g = self.__class__(create_empty_in_engine=False)
             g.graph = self.graph
             g.name = self.name
-            op = copy_graph(self, "reverse")
+            op = dag_utils.copy_graph(self, "reverse")
             g._op = op
             graph_def = op.eval(leaf=False)
             g._key = graph_def.key

--- a/python/graphscope/nx/classes/graph.py
+++ b/python/graphscope/nx/classes/graph.py
@@ -20,7 +20,6 @@
 #
 
 import copy
-import threading
 
 import orjson as json
 from networkx import freeze
@@ -369,6 +368,12 @@ class Graph(_GraphBase):
         self._saved_signature = self.signature
         self._is_client_view = False
 
+        # statically create the unload op
+        if self.op is None:
+            self._unload_op = None
+        else:
+            self._unload_op = dag_utils.unload_graph(self)
+
     def _is_gs_graph(self, incoming_graph_data):
         return (
             hasattr(incoming_graph_data, "graph_type")
@@ -376,22 +381,16 @@ class Graph(_GraphBase):
         )
 
     def __del__(self):
-        if self._session.info["status"] != "active" or self._key is None:
+        if self._key is None or self._session.disconnected:
             return
 
-        # use thread to avoid dead-lock
-        def _del(graph):
-            # cancel cache fetch future
-            if graph.cache.enable_iter_cache:
-                graph.cache.shutdown()
-            op = dag_utils.unload_graph(graph)
-            op.eval()
-            graph._key = None
+        if self.cache.enable_iter_cache:
+            self.cache.shutdown()
+        self.cache.shutdown_executor()
 
-        if not self._is_client_view:
-            t = threading.Thread(target=_del, args=(self,))
-            t.daemon = True
-            t.start()
+        if not self._is_client_view and self._unload_op is not None:
+            self._unload_op.eval()
+        self._key = None
 
     @property
     def op(self):

--- a/python/graphscope/nx/tests/classes/test_graphviews.py
+++ b/python/graphscope/nx/tests/classes/test_graphviews.py
@@ -29,7 +29,7 @@ from graphscope.nx.utils.misc import edges_equal
 
 @pytest.mark.usefixtures("graphscope_session")
 class TestReverseView(test_gvs.TestReverseView):
-    def setup(self):
+    def setup_method(self):
         self.G = nx.path_graph(9, create_using=nx.DiGraph())
         self.rv = self.G.reverse(copy=False)
         # self.rv = nx.reverse_view(self.G)
@@ -67,7 +67,7 @@ class TestReverseView(test_gvs.TestReverseView):
 
 @pytest.mark.usefixtures("graphscope_session")
 class TestToDirected(test_gvs.TestToDirected):
-    def setup(self):
+    def setup_method(self):
         self.G = nx.path_graph(9)
         self.dv = nx.to_directed(self.G)
 
@@ -87,7 +87,7 @@ class TestToDirected(test_gvs.TestToDirected):
 
 @pytest.mark.usefixtures("graphscope_session")
 class TestToUndirected(test_gvs.TestToUndirected):
-    def setup(self):
+    def setup_method(self):
         self.DG = nx.path_graph(9, create_using=nx.DiGraph())
         self.uv = nx.to_undirected(self.DG)
 

--- a/python/requirements.txt
+++ b/python/requirements.txt
@@ -3,7 +3,7 @@ gremlinpython==3.6.3rc1
 grpcio>=1.49
 grpcio-tools>=1.49
 kubernetes>=24.2.0
-msgpack
+msgpack>=1.0.5
 mypy-protobuf>=3.4.0
 nest_asyncio
 networkx==2.8.0;python_version>="3.8"

--- a/python/setup.py
+++ b/python/setup.py
@@ -193,6 +193,7 @@ def parsed_package_data():
     return {
         "graphscope": [
             "VERSION",
+            "proto/*.pyi",
         ],
     }
 

